File: myapp/lib/Models/Etudiant.dart

class Etudiant{
  String N_mat="";
  String Nom="";
  String Prenom="";
  String Classe="";
  String Phone="";
  String Mail="";
  String Parcours="";

  Etudiant(this.N_mat, this.Nom,this.Prenom, this.Classe, this.Parcours, this.Phone, this.Mail);

  Map toJson() => {
    'N_mat':N_mat,
    'Nom':Nom,
    'Prenom':Prenom,
    'Niveau':Classe,
    'Parcours':Parcours,
    'Phone':Phone,
    'Mail':Mail
  };

}

File: gestion_note/app/Models/Etudiant.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Etudiant extends Model
{
    use HasFactory;

    protected $table = "etudiant";

    protected $primaryKey = 'N_mat';
    public $incrementing = false;
    protected $fillable = ['Nom', 'Prenom', 'Niveau', 'Parcours', 'Phone', 'Mail'];

    public function note()
    {
        return $this->hasMany(Note::class);
    }
}
